DigitalLearningSolutions.Data/Services/ActivityService.cs

```diff
@@ -0,0 +1,64 @@
+﻿namespace DigitalLearningSolutions.Data.Services
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using DigitalLearningSolutions.Data.DataServices;
+    using DigitalLearningSolutions.Data.Models.TrackingSystem;
+
+    public interface IActivityService
+    {
+        public IEnumerable<MonthOfActivity> GetRecentActivity(int centreId);
+    }
+
+    public class ActivityService : IActivityService
+    {
+        private readonly IActivityDataService activityDataService;
+        private readonly IClockService clockService;
+
+        public ActivityService(IActivityDataService activityDataService, IClockService clockService)
+        {
+            this.activityDataService = activityDataService;
+            this.clockService = clockService;
+        }
+
+        public IEnumerable<MonthOfActivity> GetRecentActivity(int centreId)
+        {
+            var endTime = clockService.UtcNow;
+            var startTime = endTime.AddYears(-1);
+
+            var activityData = activityDataService.GetActivityInRangeByMonth(centreId, startTime, endTime).ToList();
+
+            var monthsThisYear = Enumerable.Range(1, endTime.Month).Select(
+                m => new MonthOfActivity
+                {
+                    Year = endTime.Year,
+                    Month = m
+                }
+            );
+            var monthsLastYear = Enumerable.Range(startTime.Month, 13 - startTime.Month).Select(
+                m => new MonthOfActivity
+                {
+                    Year = startTime.Year,
+                    Month = m
+                }
+            );
+

```

Comment:
I think we should try and make this more generic and support creating monthly slots between any startTime & endTime. This could live in a separate helper, which we can use to grab slots that we can then map to the Usage stats data we need.

Maybe something like getting months between start date & end date and then creating an Enumerable.Range(startTime.Month, monthsBetweenDates) that we can then handle. (I'm assuming the default DateTime library has no ways of doing this). Or alternatively, we could keep adding 1 month until date > endDate?